dockstore-webservice/src/main/java/io/dockstore/webservice/resources/DockerRepoTagResource.java

```diff
@@ -246,4 +258,31 @@ private Tool findToolByIdAndCheckToolAndUser(Long toolId, User user) {
         checkUser(user, tool);
         return tool;
     }
+
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("{containerId}/tags/{tagId}/sourcefiles")
+    @Operation(operationId = "getTagsSourcefiles", description = "Retrieve sourcefiles for a container's version", security = @SecurityRequirement(name = OPENAPI_JWT_SECURITY_DEFINITION_NAME))
+    public SortedSet<SourceFile> getTagsSourceFiles(@ApiParam(hidden = true) @Auth Optional<User> user,
+            @Parameter(name = "containerId", description = "Container to retrieve the version from", required = true, in = ParameterIn.PATH) @PathParam("containerId") Long containerId,
+            @Parameter(name = "tagId", description = "Tag to retrieve the sourcefiles from", required = true, in = ParameterIn.PATH) @PathParam("tagId") Long tagId,
+            @Parameter(name = "fileTypes", description = "List of file types to filter sourcefiles by") @QueryParam("fileType") List<DescriptorLanguage.FileType> fileTypes) {
+        Tool tool = toolDAO.findById(containerId);
+        checkEntry(tool);
+        checkOptionalAuthRead(user, tool);
+
+        Set<Tag> tags = tool.getWorkflowVersions();
```

Comment:
You should ideally call `TagDAO.findById()` instead. That way you will fetch only one version from the DB and not the 600+ versions some repos have.

The problem is that you'll also need to validate that `tagId` belongs to `containerId`, otherwise you could end up fetching a tag for a different container. If you fetch all of the parent's versions to verify, then you're back to square one.

1. You make Version.entry field public, so you can do the check.
2.  You write a new query that fetches a version given a containerId and tagId as params.

The second option would execute the fastest, but would be a little more work.